packages/api/src/middleware/validation/checkProductValidity.js

```diff
@@ -62,18 +82,39 @@ export function checkProductValidity() {
 
     const trx = await transaction.start(ProductModel.knex());
 
-    // Check name uniqueness
     try {
-      const { farm_id } = req.headers;
-      const { product_id } = req.params;
-      let { type, name } = req.body;
-
       if (product_id) {
         const currentRecord = await ProductModel.query(trx).findById(product_id);
+        // Prevent changing type for now, prevents orphan task type products
+        if (type && type != currentRecord.type) {
+          return res.status(400).send('cannot change product type');
+        }
+
         type = type ?? currentRecord.type;
         name = name ?? currentRecord.name;
       }
 
+      if (!name) {
+        return res.status(400).send('new product must have name');
+      }
+
+      if (!type) {
+        return res.status(400).send('new product must have type');
+      }
+
+      const nonModifiableAssets = Object.values(taskTypeProductMap).filter((productType) => {
+  return productType !== taskTypeProductMap[type];
+});
```

Comment:
Why wasn't this formatted? Interesting...

Reply:
I accepted Github commit suggestion -- no pre-commit. :(

Reply:
Oh I see!